GRDB/Core/DatabaseCollation.swift

```diff
@@ -50,6 +50,6 @@ extension DatabaseCollation: Hashable {
     /// :nodoc:
     public static func == (lhs: DatabaseCollation, rhs: DatabaseCollation) -> Bool {
         // See https://www.sqlite.org/c3ref/create_collation.html
-        return sqlite3_stricmp(lhs.name, lhs.name) == 0
+        return sqlite3_stricmp(lhs.name, rhs.name) == 0
```

Comment:
Why not also hash on the name, instead of 0?

Reply:
Because we have to honor the Hashable/Equatable rule: if a equals b, then a and b must have the same hash.

SQLite only defines collation equality in https://www.sqlite.org/c3ref/create_collation.html:

> Collation names that compare equal according to [sqlite3_strnicmp()](https://www.sqlite.org/c3ref/stricmp.html) are considered to be the same name.

I don't know exactly how this C function is implemented. I also know that native Swift String comparisons handles unicode in a way that most certainly does not match the SQLite comparison.

That's why DatabaseCollation implementation of `==` uses the C SQLite function.

And the easy way to assert the Hashable/Equatable rule is to have all values have the same hash :-)

Reply:
A classic misinterpretation of Set and Dictionary is that they consider two values or keys that have the same hash to be the same. This is not true! Hashing is only an optimization. When all values have the same hash, sets and dictionaries are not efficient. But the invariant remain: values and keys are uniqued according to Equatable, not Hashable.

In our specific case, we don't have to expect many operations on collations using GRDB apis. So we don't care if all collations have the same hash.